pkg/kubelet/client/kubelet_client.go

```diff
@@ -93,7 +93,8 @@ func (c *KubeletClientConfig) transportConfig() *transport.Config {
 			KeyFile:  c.KeyFile,
 			KeyData:  c.KeyData,
 		},
-		BearerToken: c.BearerToken,
+		BearerToken:       c.BearerToken,
+		ConnectionTimeout: 7 * time.Second,
```

Comment:
@liggitt @sttts Thanks for this. I agree that this behaviour should be configurable, so I've added `connection-timeout` option in the `cluster` section of kubeconfig, with a default value of 7 seconds. I've verified this works correctly.

I've also corrected how this was being applied to the wrong client.